enhancements/results-forwarding.md

```diff
@@ -0,0 +1,379 @@
+  and doesn't provide all the assurances that one would expect:
+  
+  - It's non-trivial to do disaster recovery: An object snapshot contains
+    CRD metadata which is hard to replicate (e.g. object UUID) and is irrelevant
+    for compliance checks. Full etcd restore is a tedious and complicated process [1].
+
+  - Values are limited to a size of 1.5Mb which makes evidence storage
+    tricky and expensive. e.g. we currently use one PVC per `ComplianceScan`
+    for raw results which may be considered evidence.
+
+  - The current CRDs may be very verbose, which may result in big objects, which
+    may incur a penalty in performance and scalability for Kubernetes. [2]
+
+While projects such as StackRox [3] have successfully integrated Compliance Operator
+via CRDs, having a more lightweight approach 
+
+By sending results and evidences to a central store, we'll be able to:
+
+* View Compliance results and suggested fixes for multiple clusters in one place.
+
+* Address the aforementioned limitations with Etcd by choosing an alternative
+  store. e.g. a relational database or a cloud-managed database.
+
+The Compliance Operator also stores raw results (currently ARF) in a PVC by default.
+While this is useful, it has proven problematic in multiple deployments given that
+not everybody has access to persistent storage in their clusters.
+
+ARF files contain a lot of information about the state of the system where the
+compliance scan was effectuated. We could take the view that this is "evidence"
+gathered for the system in check. With this in mind, we can generalize and
+expand the concept to allow for forwarding this evidence to an alternative
+object store, e.g. S3.
+
+In the future, we could expand the "evidence" gathered, e.g. for Kubernetes
+compliance scans we could simply compress the gathered objects the
+`api-resource-collector` downloads and send that too.
+
+### Goals
+
+* Provide a generic way to implement compliance result forwarding for Compliance Operator.
+  This should be flexible enough to accommodate the current implementation (CRDs and PVCs),
+  while allowing for custom forwarder implementations.
+
+* Implement and define a stable API for forwarder implementations.
+
+* Implement result/remediation CRDs and ARF PVC as reference implementations of the forwarding
+  model.
+
+### Non-Goals
+
+* Deprecate the current CRD/PVC approach: There is still a use-case for this.
+
+* Expand the "evidence" that's currently gathered by the Compliance Operator. This is
+  needed but should be done as part of a separate initiative.
+
+## Proposal
+
+In short, the proposal is to have Compliance Operator *always* forward Compliance results (suggested
+remediations being part of this), as well as evidence (e.g. ARF results). This will require
+changing the **aggregator** to always forward via a GRPC to a configured implementation. The
+**resultserver** then becomes subject to the evidence forwarding provider and will be renamed
```

Comment:
Is the resultserver supposed to consume results via the GRPC API?

Reply:
I've been thinking about this, and I think we resultserver should keep receiving results as it current is. GRPC doesn't work well for bigger files (e.g. ARF files can get a little big)... HTTP is still a better choice for this kind of payload. However, the resultserver can then forward the results using an alternative mechanism, e.g. `s3`... or simply keep using a PVC. These providers need to be configurable.